### ISSUE TITLE
False positive CA2007 when await using an IAsyncDisposable

### ISSUE THREAD

**Issue by tom-englert** [dotnet/roslyn-analyzers#7185]
### Analyzer
**Diagnostic ID**: [CA2007](https://docs.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca2007): `Consider calling ConfigureAwait on the awaited task`
 
### Analyzer source
**SDK**: [Built-in CA analyzers in .NET 6 SDK or later](https://docs.microsoft.com/dotnet/fundamentals/productivity/code-analysis)

**Version**: [SDK 8.0.100](https://dotnet.microsoft.com/download/dotnet/8.0)

### Describe the bug

an `await using` statement on an `IAsyncDisposable` object triggers a false positive CA2007:

### Steps To Reproduce
1. Make sure all analyzers are enabled in your project.
2. Add the following line of code:
```c#
await using var stream = System.IO.File.OpenRead("some file path");
```
3. See warning: ` warning CA2007: Consider calling ConfigureAwait on the awaited task`

### Expected behavior

No CA2007 is raised on `await using` when the target object does not derive from `Task`

### Actual behavior

CA2007 is raised on `await using` even when the target object does not derive from `Task`

**Comment by tom-englert** [comment]
Is related to #5712
